usaspending_api/etl/management/commands/load_transactions_in_delta.py

```diff
@@ -225,10 +226,15 @@ def award_id_lookup_pre_delete(self):
         #       returned by a SELECT subquery inside the 'IN').  Thus, this code should return a dataframe directly,
         #       create a temporary view from the dataframe in award_id_lookup_post_delete, and use that temporary
         #       view to either do a subquery in the 'IN' clause or to JOIN against.
-        possibly_modified_award_ids = [str(row["award_id"]) for row in self.spark.sql(sql).collect()]
+        if table_exists:
```

Comment:
You can reduce the complexity of this by moving the contents inside the if block out and changing the predicate to if not table_exists...raise exception.

```
if not table_exists:
     raise Exception(f"Table: int.{self.etl_level} does not exist.")
 possibly_modified_award_ids = [str(row["award_id"]) for row in self.spark.sql(sql).collect()]
```

Reply:
Done.